bin/internal/update_dart_sdk.ps1

```diff
@@ -20,14 +20,37 @@ $cachePath = "$flutterRoot\bin\cache"
 $dartSdkPath = "$cachePath\dart-sdk"
 $dartSdkLicense = "$cachePath\LICENSE.dart_sdk_archive.md"
 $engineStamp = "$cachePath\engine-dart-sdk.stamp"
-$engineRealm = (Get-Content "$flutterRoot\bin\internal\engine.realm")
 
-if (Test-Path "$flutterRoot\bin\internal\engine.version") {
-    $engineVersion = (Get-Content "$flutterRoot\bin\internal\engine.version")
+# Test for fusion repository
+if ((Test-Path "$flutterRoot\DEPS" -PathType Leaf) -and (Test-Path "$flutterRoot\engine\src\.gn" -PathType Leaf)) {
+  # Calculate the engine hash from tracked git files.
+    $branch = (git -C "$flutterRoot" rev-parse --abbrev-ref HEAD)
+    if ($null -eq $Env:LUCI_CONTEXT) {
+        $engineVersion = (git -C "$flutterRoot"  merge-base HEAD upstream/master)
+    } else {
+        $engineVersion = (git -C "$flutterRoot" rev-parse HEAD)
+    }
+
+    if (($branch -ne "stable" -and $branch -ne "beta")) {
+        # Write the engine version out so downstream tools know what to look for.
+        $engineVersion | Out-File -FilePath "$flutterRoot\bin\internal\engine.version"
+
+        # The realm on CI is passed in.
+        if ($Env:FLUTTER_REALM) {
+            $Env:FLUTTER_REALM | Out-File -FilePath "$flutterRoot\bin\internal\engine.realm"
+            $engineRealm = "$Env:FLUTTER_REALM"
+        } else {
+            $engineRealm = (Get-Content "$flutterRoot\bin\internal\engine.realm")
```

Comment:
This needs to check whether the `engine.realm` file exists.

Without that I get an infinite loop of Dart SDK update failures if I run `bin/flutter` on a clean Flaux repository on Windows without LUCI environment variables.

(On Linux the `cat bin/internal/engine.realm` done by `update_dart_sdk.sh` also fails on a fresh clone of Flaux.  The update script prints an error message but then proceeds to download the Dart SDK.  The Linux script should also check for the presence of `engine.realm` in order to avoid printing the error.)

Reply:
I can add that since it should be removed from tracking in the git repo.